cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -108,12 +110,45 @@ public void addTicket(final Ticket ticket) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public boolean deleteTicket(final String ticketId) {
         if (StringUtils.isBlank(ticketId)) {
             return false;
         }
-        return this.serviceTicketsCache.remove(ticketId) || this.ticketGrantingTicketsCache.remove(ticketId);
+
+        final Ticket ticket = getTicket(ticketId);
+        if (ticket == null) {
+            return false;
+        }
+
+        if (ticket instanceof TicketGrantingTicket) {
+            logger.debug("Removing ticket [{}] and its children from the registry.", ticket);
+            return deleteTicketAndChildren((TicketGrantingTicket) ticket);

```

Comment:
I agree too at a theorical level, though my concern is that it may generate many unnecessary calls (as the ST is either expired or was explicitly removed at the ST validation). Pinging @mmoayyed to be sure we all agree.